codex/rest/api.nim

```diff
@@ -93,6 +93,35 @@ proc formatPeerRecord(peerRecord: PeerRecord): JsonNode =
   }
   return jobj
 
+proc formatManifestBlock(store: BlockStore, cid: Cid): Future[?JsonNode] {.async.} =
+  without blk =? await store.getBlock(cid):
+    return JsonNode.none
+
+  without manifest =? Manifest.decode(blk):
+    return JsonNode.none
+
+  let jobj = %*{
+    "cid": $cid,
+    "rootHash": $manifest.cid,
+    "originalBytes": $manifest.originalBytes,
+    "blockSize": $manifest.blockSize,
+    "protected": $manifest.protected
```

Comment:
`%*` applies a `%` to each element in the object. In this way, we let the json serializer handle the value serialization. Some elements, like integers or bools for example, don't need to be strings. If the `$` from each element is removed, we can let the serializer handle correct serialization without the need to figure that out each and every time a new `JsonObject` is created using `%*`:
```suggestion
    "cid": cid,
    "rootHash": manifest.cid,
    "originalBytes": manifest.originalBytes,
    "blockSize": manifest.blockSize,
    "protected": manifest.protected
```
If the compiler complains that `%` doesn't exist for a type, an override can be provided in `rest/json.nim`, eg
```nim
func `%`*(obj: Cid): JsonNode =
  % $obj
```

Reply:
Even better actually, is simply marking the fields on the `Manifest` type as `{.serialize.}` and applying `%` to the manifest object. It will get serialized automatically, eg:
```nim
let manifest: Manifest = getManifest()
%manifest # produces JsonObject
# OR
manifest.toJson # produces string, equivalent to $ %manifest

However, if you don't want to mark the the `Manifest` type with `{.serialize.}` pragmas, then an alternative method is to create a type, eg `RestManifest` that is specific for this rest purpose and serialize that, eg (pseudocode):

```nim
#
# rest/json.nim
#
type RestManifest* = object
  rootHash* {.serialize.}: Cid
  originalBytes* {.serialize.}: UInt256
  blockSize* {.serialize.}: UInt256
  protected* {.serialize.}: bool

proc init*(_: type RestManifest, manifest: Manifest): RestManifest =
  RestManifest(
    rootHash: manifest.rootHash,
    originalBytes: manifest.originalBytes,
    blockSize: manifest.blockSize,
    protected: manifest.protected
  )

#
# Usage in rest/api.nim
#
let restManifest = RestManifest.init(manifest)
%restManifest # serialized rest manifest (JsonObject)
# OR
restManifest.toJson # equivalent to $ %restManifest (string)
```

Reply:
Thanks very much for stickling. This input has enabled me to clean up a bunch of serializing garbage code.